### ISSUE TITLE
Support `explain` diagnostic command

### ISSUE THREAD

**Issue by AlekSi** [FerretDB/FerretDB#167]
After #372.

We should implement `explain` command for query commands like `find`, `findAndModify`, `count`, etc.

Non-query commands like `update`, `delete`, etc are not in scope.
Aggregation pipeline is not in scope.
Tigris is not in scope.

See:
* https://www.mongodb.com/docs/manual/reference/command/explain/
* https://www.mongodb.com/docs/manual/reference/explain-results/
* https://www.postgresql.org/docs/current/sql-explain.html

We don't have to replicate MongoDB output precisely; instead, we will use what `EXPLAIN (FORMAT JSON)` would return. But some similarity is needed.

* [x] convert returned JSON document into `types.Document` of the same form;
* [x] implement `explain` command:
  * [x] `verbosity` parameter should be ignored;
  * [x] `explainVersion`, `command`, `serverInfo` fields should be returned as in MongoDB;
  * [x] `queryPlanner` should contain converted `EXPLAIN (FORMAT JSON)` results;
  * [x] `executionStats` and `serverParameters` should be absent;
* [x] change `fetch` method to run `EXPLAIN (VERBOSE true, FORMAT JSON) <query>` instead of the query itself if parameter is given;
* [x] write integration test;
* [x] refactor `fetch` method (split it into two?) if it seems to be needed on review.

**Comment by seeforschauer** [comment]
To be added in test when all commands will be implemented:
```go
"FailedToParse": {
			command: bson.D{
				{
					"explain", bson.D{
						{"query", 1},
						{"update", bson.D{{"$set", bson.D{{"v", 43.13}}}}},
						{"upsert", true},
					},
				},
				{"verbosity", "queryPlanner"},
			},
			err: &mongo.CommandError{
				Code:    59, // (Location40415) 9 FailedToParse
				Message: "Explain failed due to unknown command: query",
				Name:    "CommandNotFound",
			},
		},
```

**Comment by seeforschauer** [comment]
On the query we can return the result whereas MongoDB returns an error. 
```go
"UnknownField": {
	command: bson.D{
		{
			"explain", bson.D{
				{"find", collectionName},
				{"query", bson.D{{"value", bson.D{{"$type", "array"}}}}},
			},
		},
		{"verbosity", "queryPlanner"},
	},
	err: &mongo.CommandError{
		Code:    40415, // (Location40415) 9 FailedToParse
		Message: "BSON field 'FindCommandRequest.query' is an unknown field.",
		Name:    "Location40415",
	},
},
```
